### ISSUE TITLE
Make an update site for ImageJ

### ISSUE THREAD

**Issue by imagejan** [biocompibens/SME#1]
Thank you for your great contribution!

I suggest setting up an [update site](https://imagej.net/Update_Sites) for your plugin, to make installation as easy as possible for users.

**Comment by ctrueden** [comment]
I strongly second this idea. In your Nature article, you write:

> Finally, whatever the methodological drawbacks, there seems to also be an accessibility issue as, to our knowledge, the only tool made easily available to end users is the extended depth of field (EDF) method by Aguet et al. implemented as a plugin in the popular software programme NIH Fiji/ImageJ.

But there are 100s/1000s of other ImageJ plugins not distributed with Fiji, too. It is very difficult to even know what all is available, much less do a thorough comparison. To address this accessibility issue, the ImageJ community [recommends two possible solutions](http://imagej.net/Distribution): 1) distribute your plugin with Fiji; or 2) create an update site. The alternative—simply distributing your plugin as a ZIP file which must be manually installed—is merely adding to the problem you lament. See the [Distribution page](http://imagej.net/Distribution) for further details.

**Comment by erexhepa** [comment]
Yes the update site is a very good idea to facilitate maintenance of the plugin and making the process as transparent as possible to the end user. 

At the time when were discussing this there was the migration from ImageJ1 to ImageJ2 and it was not clear whether the update site mechanism was compatible accros version. I haven't followed the ongoing migration work and development of ImageJ2 but from what I remember there were discussions about substancial changes on the structure of the IJ core package and the update site was going to be available only for ImageJ2.

**Comment by ctrueden** [comment]
> the update site was going to be available only for ImageJ2.

That is true. ImageJ 1.x does not support update sites, and it is highly unlikely it ever will.

If supporting ImageJ 1.x is a hard requirement for you (though I would ask "why?"), you could offer both: an update site for IJ2 users, and a downloadable ZIP for people who still use IJ1.

**Comment by erexhepa** [comment]
Thanks for your comments. 

I'm not sure if it is a hard requirement. Since ImageJ2 was still an ongoing work we developed the plugin targeting ImageJ1. If ImageJ2 is FULLY backward compatible then it should be OK. This is the only limitation I can see so far. It has to be tested.

**Comment by ctrueden** [comment]
> If ImageJ2 is FULLY backward compatible then it should be OK. This is the only limitation I can see so far.

Well, we go so far as to use the ImageJ 1.x user interface to ensure this.

But it depends how strictly you mean "FULLY". If you mean really truly 100% of all things that worked in ImageJ 1.x continue to work in ImageJ2, then no. Not even ImageJ 1.x is FULLY compatible with ImageJ 1.x, because there are regressions and changes—things that used to work stop working. See https://github.com/imagej/ij1-tests for some examples.

Pragmatically speaking, the main thing ImageJ 1.x can do that ImageJ2 cannot right now is dynamic reloading of plugins—i.e., the "Refresh Menus" and "Compile and Run" commands do not currently work in ImageJ2. But I do not think you need those for your use case, since you ship binaries.

> It has to be tested.

Totally agreed! :+1:

**Comment by erexhepa** [comment]
Ok then it should be fine. I will give it a go and test it.
